matminer/featurizers/composition.py

```diff
@@ -254,6 +255,83 @@ def citations(self):
     def implementors(self):
         return ["Jiming Chen", "Logan Ward", "Anubhav Jain", "Alex Dunn"]
 
+class Meredig(BaseFeaturizer):
+    """
+    Class to calculate features as defined in Meredig et. al.
+    """
+
+    def __init__(self):
+        self.data_source = MeredigData()
+
+        self.features = [e.strip()+" fraction" for e in self.data_source.element_names]
+        self.features = self.features+["mean AtomicWeight", "mean Column", "mean Row", "range AtomicNumber", "mean AtomicNumber",
+        "range AtomicRadius", "mean AtomicRadius", "range Electronegativity", "mean Electronegativity", "mean NsValence",
+        "mean NpValence", "mean NdValence", "mean NfValence", "fraction NsValence",
+        "fraction NpValence", "fraction NdValence", "fraction NfValence"]
+        # Initialize stats computer
+        self.pstats = PropertyStats()
+
+    def featurize(self, comp):
+        """
+        Get elemental property attributes
+
+        Args:
+            comp: Pymatgen composition object
+
+        Returns:
+            all_attributes: Specified property statistics of features
+        """
+
+        #First 103 features are element fractions
+        n_elements = 103
+        element_fractions = [0] * n_elements
+        el_list = list(comp.element_composition.fractional_composition.items())
+        for el in el_list:
+            obj = el
+            atomic_number_i = obj[0].number - 1
+            element_fractions[atomic_number_i] = obj[1]
+
+        #Calculate the remaining features
+        # Get the element names and fractions
+        elements, fractions = zip(*comp.element_composition.items())
+        attributes = [0] * (len(self.features) - n_elements)
+
+        #Total valence electrons, stored so we can calculate orbital fractions
+        valence_sum = 0
+        for i,feat in enumerate(self.features[n_elements:]):
```

Comment:
Suggestion, not required change: The code here is a little more complex than necessary. Rather than having a single loop that has different behavior on different iterations, consider splitting this into 2 or more loops. Further, determining which attribute to compute based on the name is also rather complicated. 

Because the order of the attributes does not matter for this featurizer, a simpler/faster algorithm could be:

1. Loop over properties that are averaged, compute averages and add to list
2. Loop over "range" properties, compute/add to list
3. Call out to `ValenceOrbital` for the remaining properties

This would also allow you to skip splitting strings and the `calc_stat` operation.

Reply:
I split out the valence orbital calculations, and the loop is now only 4 lines.

For the moment it's still calculating statistics/attributes based on string splitting - do you think in the new form that's still too complex?